File: ObjectAvoidance/java/src/main/java/elite/math/trig/Arccotangent.java
package elite.math.trig;

import elite.math.trig.triangle.Adjacent;
import elite.math.trig.triangle.Opposite;

/**
 *
 * @author CFerg (Elite)
 */
public class Arccotangent {

    /**
     *
     */
    public double arccotangent;
    
    /**
     *
     * @param o
     * @param a
     */
    public Arccotangent(Opposite o, Adjacent a){
        this.arccotangent = (o.opposite / a.adjacent);
    }
    
    /**
     *
     * @param cot
     */
    public Arccotangent(Cotangent cot){
        this.arccotangent = (1 / cot.cotangent);
    }
    
    /**
     *
     * @param tan
     */
    public Arccotangent(Tangent tan){
        this.arccotangent = tan.tangent;
    }
}

File: ObjectAvoidance/src/elite/math/trig/Arccotangent.kt
package elite.math.trig

import elite.math.trig.triangle.Adjacent
import elite.math.trig.triangle.Opposite

/**
 *
 * @author CFerg (Elite)
 */
class Arccotangent {

    /**
     *
     */
    var arccotangent: Double = 0.toDouble()

    /**
     *
     * @param o
     * @param a
     */
    constructor(o: Opposite, a: Adjacent) {
        this.arccotangent = o.opposite / a.adjacent
    }

    /**
     *
     * @param cot
     */
    constructor(cot: Cotangent) {
        this.arccotangent = 1 / cot.cotangent
    }

    /**
     *
     * @param tan
     */
    constructor(tan: Tangent) {
        this.arccotangent = tan.tangent
    }
}